lib/GADS/Role/Purgable.pm

```diff
@@ -0,0 +1,30 @@
+package GADS::Role::Purgable;
+
+use strict;
+use warnings;
+
+use MooX::Types::MooseLike::Base qw(ArrayRef);
+
+use Moo::Role;
+
+has value_fields => (
+    is      => 'lazy',
+    isa     => ArrayRef,
+    builder => sub { ['value']; }
+);
+
+sub purge {
+    my ($self,$user) = @_;
+    
+    my @field = @{$self->value_fields} or panic __"No valuefield defined";
+    my $schema = $self->result_source->schema;
+
+    my $purge_needed = grep { $self->$_ } @field;
+    $purge_needed = 1 if ref($self) eq "Gads::Schema::Result::File";
```

Comment:
Don't believe this is needed now it's using "around" but not certain so leaving it in for now.

Reply:
Instead of this, I would create a new method called `is_purged` and override it in `GADS::Datum::File`